From 7dd49a45955f78580e9cff600e5882cb28a6940a Mon Sep 17 00:00:00 2001
From: jcn <jcn@8da58560-a4e7-4996-a0c2-a735b94b261c>
Date: Mon, 20 Oct 2008 17:29:04 +0000
Subject: [PATCH] move-copy-sheet.xml: use line-count="7".  Fixes n#548091.

   * Modified     toolkit/workben/layout/move-copy-sheet.xml
   * Modified     sc/uiconfig/layout/move-copy-sheet.xml

fixed compiler warnings

   * Modified     vcl/source/control/lstbox.cxx    

git-svn-id: svn+ssh://svn.services.openoffice.org/ooo/cws/layoutdialogs2@264352 8da58560-a4e7-4996-a0c2-a735b94b261c

layoutdialogs2: vcl: respect line-count also for non-dropdown listbox.

   * Modified     vcl/inc/vcl/lstbox.hxx
   * Modified     vcl/source/control/lstbox.cxx    

git-svn-id: svn+ssh://svn.services.openoffice.org/ooo/cws/layoutdialogs2@262552 8da58560-a4e7-4996-a0c2-a735b94b261c
---
 sc/uiconfig/layout/move-copy-sheet.xml     |    2 +-
 toolkit/workben/layout/move-copy-sheet.xml |    2 +-
 vcl/inc/vcl/lstbox.hxx                     |    1 +
 vcl/source/control/lstbox.cxx              |   14 ++++++--------
 4 files changed, 9 insertions(+), 10 deletions(-)

diff --git sc/uiconfig/layout/move-copy-sheet.xml sc/uiconfig/layout/move-copy-sheet.xml
index 6e68b5a..7c4af29 100644
--- sc/uiconfig/layout/move-copy-sheet.xml
+++ sc/uiconfig/layout/move-copy-sheet.xml
@@ -9,7 +9,7 @@
         <fixedtext id="FT_DEST" _label="To ~document"/>
         <listbox has_border="true" dropdown="true" id="LB_DEST" tabstop="true"/>
         <fixedtext id="FT_INSERT" _label="~Insert before"/>
-        <listbox has_border="true" id="LB_INSERT" tabstop="true" string-item-list=" : : : : " />
+        <listbox has_border="true" id="LB_INSERT" tabstop="true" line-count="7" string-item-list=" : : : : " />
         <checkbox id="BTN_COPY" tabstop="true" _label="~Copy"/>
 	<dialogbuttonhbox border="5" spacing="5">
 	    <flow/>
diff --git toolkit/workben/layout/move-copy-sheet.xml toolkit/workben/layout/move-copy-sheet.xml
index 6e68b5a..7c4af29 100644
--- toolkit/workben/layout/move-copy-sheet.xml
+++ toolkit/workben/layout/move-copy-sheet.xml
@@ -9,7 +9,7 @@
         <fixedtext id="FT_DEST" _label="To ~document"/>
         <listbox has_border="true" dropdown="true" id="LB_DEST" tabstop="true"/>
         <fixedtext id="FT_INSERT" _label="~Insert before"/>
-        <listbox has_border="true" id="LB_INSERT" tabstop="true" string-item-list=" : : : : " />
+        <listbox has_border="true" id="LB_INSERT" tabstop="true" line-count="7" string-item-list=" : : : : " />
         <checkbox id="BTN_COPY" tabstop="true" _label="~Copy"/>
 	<dialogbuttonhbox border="5" spacing="5">
 	    <flow/>
diff --git vcl/inc/vcl/lstbox.hxx vcl/inc/vcl/lstbox.hxx
index 2390ab9..bfcac48 100644
--- vcl/inc/vcl/lstbox.hxx
+++ vcl/inc/vcl/lstbox.hxx

@@ -57,6 +57,7 @@ private:
 	BOOL						mbDDAutoSize;
 	Link						maSelectHdl;
 	Link						maDoubleClickHdl;
+    USHORT mnLineCount;
 
 //#if 0 // _SOLAR__PRIVATE
 private:
diff --git vcl/source/control/lstbox.cxx vcl/source/control/lstbox.cxx
index b22abb5..8a510d4 100644
--- vcl/source/control/lstbox.cxx
+++ vcl/source/control/lstbox.cxx
@@ -604,18 +604,18 @@ BOOL ListBox::IsDDAutoWidthEnabled() const
 
 void ListBox::SetDropDownLineCount( USHORT nLines )
 {
+    mnLineCount = nLines;
 	if ( mpFloatWin )
-		mpFloatWin->SetDropDownLineCount( nLines );
+		mpFloatWin->SetDropDownLineCount( mnLineCount );
 }
 
 // -----------------------------------------------------------------------
 
 USHORT ListBox::GetDropDownLineCount() const
 {
-	USHORT nLines = 0;
-	if ( mpFloatWin )
-		nLines = mpFloatWin->GetDropDownLineCount();
-	return nLines;
+    if ( mpFloatWin )
+        return mpFloatWin->GetDropDownLineCount();
+	return mnLineCount;
 }
 
 // -----------------------------------------------------------------------
@@ -1260,9 +1260,7 @@ Size ListBox::CalcMinimumSize() const
 {
 	Size aSz;
 	if ( !IsDropDownBox() )
-	{
-		aSz = mpImplLB->CalcSize( mpImplLB->GetEntryList()->GetEntryCount() );
-	}
+        aSz = mpImplLB->CalcSize (mnLineCount ? mnLineCount : mpImplLB->GetEntryList()->GetEntryCount());
 	else
 	{
 		aSz.Height() = mpImplLB->CalcSize( 1 ).Height();
-- 
1.6.3.3

